### DISCUSSION TITLE
Scoped flags/env vars

### DISCUSSION THREAD

**Discussion by Disane87** [oclif/oclif#1318] [Q&A] [answered]
Thank you for this awesome project! Really love this.

Currently I'm evaluating this to use it in my cli application with can be enriched by user plugins (it's essentially an universal webscraper for pdf documents to download them and out them in a DMS, because i.e. amazon and other provider/vendors only provide invoices only in their customer portal).

Now I want to enforce a use to name their flag env vars like `PLUGINNAME_VARNAME` to circumvent the potential conflicts in env vars like usernames for auth etc.

Is there any possibility to achive this out of the box or should I create a BaseClass where I encapsulate your flags and generate the name?

What would be the best practise approach in your oppinion?

**Comment by mdonnalley** [comment, marked as answer]
>Is there any possibility to achive this out of the box or should I create a BaseClass where I encapsulate your flags and generate the name?

We don't have anything that would accomplish this for you, so a base class will be your best bet.
